index.bs

```diff
@@ -2137,6 +2142,54 @@ document loading at which a navigation command will return.
 
 ### Commands ### {#module-browsingContext-commands}
 
+#### The browsingContext.activate Command ####  {#command-browsingContext-activate}
+
+The <dfn export for=commands>browsingContext.activate</dfn> command activates and focuses the given top-level browsing context.
+
+<dl>
+   <dt>Command Type</dt>
+   <dd>
+    <pre class="cddl remote-cddl">
+      browsingContext.Activate = {
+        method: "browsingContext.activate",
+        params: browsingContext.ActivateParameters
+      };
+
+      browsingContext.ActivateParameters = {
+        context: browsingContext.BrowsingContext
+      }
+    </pre>
+   </dd>
+   <dt>Result Type</dt>
+   <dd>
+    <pre class="cddl">
+     EmptyResult
+    </pre>
+   </dd>
+</dl>
+
+<div algorithm="remote end steps for browsingContext.activate">
+
+The [=remote end steps=] with |command parameters| are:
+
+1. Let |context id| be the value of the |command parameters|["<code>context</code>"] field.
+
+1. Let |context| be the result of [=trying=] to [=get a browsing context=] with |context id|.
+
+1. If |context| is not a [=top-level browsing context=], return [=error=] with [=error code=] [=invalid argument=].
+
+1. Run implementation-specific steps so that |context|'s [=traversable navigable=]'s [=system visibility state=] becomes [=visible=]. If this is not possible return [=error=] with error code [=unsupported operation=].
+
+    Note: This can have the side effect of making currently [=visible=] [=navigables=] [=hidden=].
+
+    Issue: It's not clear if we should have explicit requirements for cases like a minimized window; should that cause the window to become unminimized?
```

Comment:
https://html.spec.whatwg.org/multipage/interaction.html#page-visibility:system-visibility-state says:

> A [traversable navigable](https://html.spec.whatwg.org/multipage/document-sequences.html#traversable-navigable)'s [system visibility state](https://html.spec.whatwg.org/multipage/document-sequences.html#system-visibility-state), including its initial value upon creation, is determined by the user agent. It represents, for example, whether the browser window is minimized, a browser tab is currently in the background, or a system element such as a task switcher obscures the page.

To me, this reads as if we might already have these requirements implicitly with the current wording, and there’s no further action needed.

Reply:
I think if we're assuming that this has to adjust the underlying OS state in some cases we should call that out at least as a note. It also means we can't test this command correctly without using classic for window manipulation, or adding the window manipulation commands to BiDi.